tools/chef/roles/web-nginx-ssl-offload.json

```diff
@@ -0,0 +1,43 @@
+{
+  "name": "web-nginx-ssl-offload",
+  "chef_type": "role",
+  "json_class": "Chef::Role",
+  "description": "Nginx HTTP server role for SSL offloading",
+  "default_attributes": {
+    "nginx": {
+      "package": "nginx18",
+      "default_site_enabled": false,
+      "https_variable_emulation": false,
+      "shared_config": {
+        "ssl_offload": {
+          "type": "ssl_offload",
+          "protocols": ["https"],
+          "disable_default_location_block": true,
+          "php_support": false,
+          "locations": {
+            "/": {
+              "type": "path_not_regex",
```

Comment:
I was actually thinking of dropping the path in regex form when some nesting changes @shrikeh and I have discussed means that it will no longer be needed for the location precedence (it was previously so the \.php location wouldn't trigger as it had higher precedence)